src/main/java/org/jlab/jnp/grapes/services/JpsiTcsWagon.java

```diff
@@ -1,60 +1,82 @@
 package org.jlab.jnp.grapes.services;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+
 import org.jlab.jnp.hipo4.data.Bank;
 import org.jlab.jnp.hipo4.data.Event;
 import org.jlab.jnp.hipo4.data.SchemaFactory;
-import java.util.ArrayList;
-import java.util.HashMap;
+import org.jlab.jnp.physics.LorentzVector;
 
 /**
  * 
- * Skim for JPsi/TCS analysis group, E12-12-001.
+ * Skim for JPsi/TCS analysis group.
  *
  * Until standard filters provide access to momentum and ECAL energy.
  *
  * @author baltzell
  */
-public class JpsiTcsWagon extends Wagon {
+public class JpsiTcsWagon extends BeamTargetWagon {
+
+    boolean includeMissingJPsi = true;
 
+    static final float MIN_TAGGED_MISS_MASS = 1.50f; 
     static final float MIP_ECOUT_MAX = 0.110f;
     static final float MIP_ECIN_MAX = 0.100f;
     static final float MIP_PCAL_MAX = 0.200f;
     static final float MOM_HIGH = 2.0f;
 
     public JpsiTcsWagon(){
-        super("JpsiTcsWagon","baltzell","0.5");
+        super("JpsiTcsWagon","baltzell","0.6");
     }
 
     @Override
     public boolean init(String jsonString) {
-        System.out.println("JpsiTcsWagon READY.");
+        if (!super.init(jsonString)) {
+            includeMissingJPsi = false;
+            System.out.println(engineName+":  Missing-Jpsi Disabled.");
+        }
         return true;
     }
 
-    private HashMap<Integer,ArrayList<Integer>> mapByIndex(Bank pindexBank) {
-        HashMap<Integer,ArrayList<Integer>> map=new HashMap<>();
-        for (int ii=0; ii<pindexBank.getRows(); ii++) {
-            final int pindex = pindexBank.getInt("pindex",ii);
-            if (!map.containsKey(pindex)) map.put(pindex,new ArrayList<>());
-            map.get(pindex).add(ii);
+    /**
+     * Stepan's request for electron in FT and a proton with minimum missing mass.
+     * @param event
+     * @param factory
+     * @return 
+     */
+    public boolean isMissingAndTagged(Event event, SchemaFactory factory) {
+        Bank particles = new Bank(factory.getSchema("RECFT::Particle"));
+        event.read(particles);
+        if (particles.getRows()<2) return false;
+        if (particles.getInt("pid",0) != 11) return false;
```

Comment:
in RECFT, the first particle is not the FT electron because the order is the same as in REC::Particle